dali/operators/reader/loader/loader.h

```diff
@@ -217,8 +217,8 @@ class Loader {
   void PrepareMetadata() {
     std::lock_guard<std::mutex> l(prepare_metadata_mutex_);
     if (!loading_flag_) {
-      loading_flag_ = true;
       PrepareMetadataImpl();
+      loading_flag_ = true;
```

Comment:
If we're paranoid, we could insert a memory fence before setting this flag, so we're 100% sure that other threads will see all the changes before this flag is set.

Reply:
Done